Automattic/vip-go-mu-plugins
# Require current password when changing it

## Description


This PR introduces the requirement to input the current user's password when changing it on its profile page in `wp-admin`.  This has been implemented using WP actions. There is some JS code that is being used to reposition the elements in the form.

The new input is only shown when the user clicks `Set New Password`:

<img width="583" alt="Screen Shot 2021-04-08 at 4 56 30 PM" src="https://user-images.githubusercontent.com/7188409/114049115-7540f700-988b-11eb-8f02-d99dbc3a44d3.png">

<img width="749" alt="Screen Shot 2021-04-08 at 4 56 23 PM" src="https://user-images.githubusercontent.com/7188409/114049130-78d47e00-988b-11eb-97ff-c04363c79450.png">

If the user doesn't input a password, an error is thrown:

<img width="404" alt="Screen Shot 2021-04-08 at 5 01 05 PM" src="https://user-images.githubusercontent.com/7188409/114049762-02844b80-988c-11eb-8b42-1336db7b25cc.png">

If the password is not correct, there is a different error:

<img width="365" alt="Screen Shot 2021-04-08 at 5 01 33 PM" src="https://user-images.githubusercontent.com/7188409/114049843-1334c180-988c-11eb-9d57-b4ca7957ef25.png">

## Changelog Description


### Require current password for changing it

When changing its own password in the WP admin, the user will be prompted with their current password.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test


1. Check out PR.
2. Go to `wp-admin` > `Users` > `Profile`
3. Change password by clicking `Set New Password`
